src/common/colorlabels.c

```diff
@@ -132,8 +132,29 @@ typedef enum dt_colorlabels_actions_t
 
 
 
-static void _colorlabels_execute(const GList *imgs, const int labels, GList **undo, const gboolean undo_on, const int action)
+static void _colorlabels_execute(const GList *imgs, const int labels, GList **undo, const gboolean undo_on, int action)
 {
+  if(action == DT_CA_TOGGLE)
+  {
+    // if we are supposed to toggle color labels, first check if all images have that label
+    for(const GList *images = imgs; images; images = g_list_next((GList *)images))
```

Comment:
Just a nitpicking point, here I would use `GList *image` for the iterator, as it is used to get the image_id for a single image at a time. This is I think the convention used in dt.

Reply:
In fact now, the loop just below is using `images`, consistency is hard to keep :) Anyway, feel free to change both loops if you agree.